src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -84,6 +87,45 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
     Await.result(futureStore, DatabaseTimeout)
   }
 
+ /**
+   * Try to generate output for a collector call, by collecting outputs for all of its shards.
+   * It's fail-fast on shard output retrieval
+   */
+  def generateCollectorOutput(collector: CollectorKey, shards: Iterable[CallKey]): Try[CallOutputs] = Try {
+    val shardsOutputs = shards map { e =>
+      fetchCallOutputEntries(e.scope) map { _.value } get
+    }
```

Comment:
The code seems to assume the shards are being presented in the correct order for the construction of the collector array; is that a safe assumption?